### DISCUSSION TITLE
What is the correct/best way to get $q.bex/bridge in a boot file or a pinia store?

### DISCUSSION THREAD

**Discussion by richterdennis** [quasarframework/quasar#17139] [CLI - BEX mode]
What is the correct/best way to get $q.bex/bridge in a boot file or a pinia store?

When I open my bex popup i want to load the state from the background script. The only way to communicate with the background script is through the $q.bex/bridge. But I don't have access to `this.$q.bex` in a boot file or in a pinia store function and there is nothing like `useBridge`.

My best approach is something like this:
```
const globalProperties = app.config.globalProperties;
let bridge = null;

while (!bridge) {
	await wait(50);
	bridge = globalProperties.$q?.bex;
}
```

But this is ofc not a good way. How do I do this the correct way?

**Comment by Flamenco** [comment]
It appears the bridge is exposed on the window object.
This is not really documented though.

You can roll your own function:

```ts
function useBridge() {
  return window.QBexBridge
}
```

```ts
const bridge = useBridge() 
```

<img width="361" alt="Screenshot 2024-07-20 at 11 05 02 AM" src="https://github.com/user-attachments/assets/3dc6c849-2f56-47ed-ba04-392f6eb83245">

To get typescript to stop whining, try adding
```ts
declare global {
  interface Window {
    QBexBridge: any
  }
}
```

For reference, here is where the global is initialized.  It happens right before `mount` is called.

<img width="536" alt="Screenshot 2024-07-20 at 11 20 50 AM" src="https://github.com/user-attachments/assets/52c540f9-6abf-45ab-8fa2-18df06a302e9">

**Comment by derrickorama** [comment]
I ran into this same problem trying to use QBexBridge in a boot file used in a side panel. The best way I found was to listen for the "app" connection in the background script and send a message indicating QBexBridge is initialized.

In your background file:
```javascript
chrome.runtime.onConnect.addListener((port) => {
  if (port.name.startsWith('app:')) { // name will be app:<random uid>
    chrome.runtime.sendMessage('ready');
  }
});
``` 

And in your boot file:
```javascript
import { boot } from 'quasar/wrappers';

export default boot(() => {
  chrome.runtime.onMessage.addListener((message) => {
    if (message === 'ready') {
      console.log('window.QBexBridge:', window.QBexBridge);
      // do things that rely on QBexBridge
    }
  });
});
```
